src/modules/commander/PreflightCheck.cpp

```diff
@@ -382,6 +383,87 @@ static bool gnssCheck(orb_advert_t *mavlink_log_pub, bool report_fail)
 	return success;
 }
 
+static bool ekfCheck(orb_advert_t *mavlink_log_pub, bool optional, bool report_fail)
+{
+	bool success = true;
+
+	int fd1 = orb_subscribe(ORB_ID(ekf2_innovations));
+	struct ekf2_innovations_s innovations;
+	orb_copy(ORB_ID(ekf2_innovations), fd1, &innovations);
```

Comment:
What if orb_copy fails (because it's not advertised for example)? Data will be random.
What if LPE is used?